From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@sourcemage.org>
Date: Sun, 16 Jul 2023 21:54:21 +0200
Subject: [PATCH] Fix build against musl libc

The build fails because of missing off64_t definition; in order
for musl to expose off64_t, the source file must define the
_LARGEFILE64_SOURCE macro.

Origin: Source Mage
Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 absl/base/internal/direct_mmap.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/direct_mmap.h b/direct_mmap.h
index 815b8d23ba35..d34bd74c7a92 100644
--- a/absl/base/internal/direct_mmap.h
+++ b/absl/base/internal/direct_mmap.h
@@ -33,6 +33,7 @@
 #include <syscall.h>
 #endif
 
+#define _LARGEFILE64_SOURCE /* for off64_t */
 #include <linux/unistd.h>
 #include <unistd.h>
 #include <cerrno>
